04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,77 @@
+const Product = require("../models/product");
+
+const getAllProductsStatic = async (req, res) => {
+  const products = await Product.find({ price: { $gt: 30 } })
+    .sort("price")
+    .select("name price");
+  // .limit(10)
+  // .skip(1);
+
+  // throw new Error('testing async errors')
+  res.status(200).json({ products, nbHits: products.length });
+};
+
+const getAllProducts = async (req, res) => {
+  const { featured, company, name, sort, fields, numericFilters } = req.query;
+  const queryObject = {};
+
+  if (featured) {
+    queryObject.featured = featured === "true" ? true : false;
+  }
+  if (company) {
+    queryObject.company = company;
+  }
+  if (name) {
+    queryObject.name = { $regex: name, $options: "i" };
+  }
+  if (numericFilters) {
+    const operatorMap = {
+      ">": "$gt",
+      ">=": "$gte",
+      "=": "$eq",
+      "<": "$lt",
+      "<=": "$lte",
+    };
+    const regEx = /\b(<|>|<=|=|>=)\b/g;
+    let filters = numericFilters.replace(
+      regEx,
+      (match) => `-${operatorMap[match]}-`
+    );
+    const options = ["price", "rating"];
+    filters = filters.split(",").forEach((item) => {
```

Comment:
This assignment isn't needed since we are simply building out `queryObject` using the `forEach`. A `forEach` also doesn't return anything.
So it could just be:
`filters.split(",").forEach((item) => { // ...`

See:
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#return_value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#return_value)